test/Mono.Linker.Tests/TestCasesRunner/TestCaseMetadaProvider.cs

```diff
@@ -55,23 +55,62 @@ public virtual TestCaseLinkerOptions GetLinkerOptions ()
 			return tclo;
 		}
 
+#if NETCOREAPP
+		public static IEnumerable<string> GetTrustedPlatformAssemblies ()
+		{
+			if (AppContext.GetData ("TRUSTED_PLATFORM_ASSEMBLIES") is string tpaPaths)
+			{
+				foreach (var path in tpaPaths.Split(Path.PathSeparator))
+				{
+					if (Path.GetExtension (path) == ".dll")
+						yield return path;
+				}
+			}
+		}
+#endif
+
 		public virtual IEnumerable<string> GetCommonReferencedAssemblies (NPath workingDirectory)
 		{
 			yield return workingDirectory.Combine ("Mono.Linker.Tests.Cases.Expectations.dll").ToString ();
+#if NETCOREAPP
+			foreach (var path in GetTrustedPlatformAssemblies())
+			{
+				// Don't reference testcases dll, as these will be compiled dynamically.
+				if (Path.GetFileName (path) != "Mono.Linker.Tests.Cases.dll")
+					yield return path;
```

Comment:
It's a bit unfortunate, but this gives back a huge list of assemblies representing the framework and the test project dependencies (see the list: https://gist.github.com/sbomer/dc5ec8cbd2df2311817525aa2f0f882c), since the framework has been split out into many dlls for core. I could filter it down to `System.Private.CoreLib` and about 10-20 others, but that has the downside that the testcase dependency dlls need to be explicitly listed. I could add those dependencies to a the common references, or just the testcases that need them via `ReferenceAttribute`. The list I would need is approximately (in the order I encountered them):
- System.Private.CoreLib
- System.Runtime
- System
- System.ComponentModel.TypeConverter
- System.Security.Permissions (which comes from a nuget package)
- netstandard
- System.Runtime.Extensions
- System.Diagnostics.Process
- System.Console
- mscorlib
- System.Collections.NonGeneric
- System.Collections
- System.Resources.Writer
- System.Private.Uri

Whether I add these per-testcase or to the common references, I'll need to resolve them using an approach similar to the TPA list. For example, if I add `[Reference(System.Security.Permissions)]` to a testcase, then something will need to resolve that to the full path, which is something like `/Users/sven/src/dotnet/linker/test/Mono.Linker.Tests/bin/illink_Debug/netcoreapp3.0/System.Security.Permissions.dll`. Most likely this would be done in `GetReferencedAssemblies` by finding items on the TPA list that have the right filename.

What do you think? FWIW our SDK, when compiling a .NET Core project, adds references to the whole framework by default, so the current behavior isn't too different.

Reply:
Let me know after you give VS2019 a try and see if there are issues with that or if it's something on my end.  I'd like to try it locally and poke around a bit in order to get my head around it better.

Reply:
I'm not going to have a chance to try this, but it seems fine.